scripts/install_backend_python_libs.py

```diff
@@ -29,19 +30,53 @@
 
 
 def _normalize_python_library_name(library_name):
-    """Returns a normalized (lowercase) version of the python library name.
-
-    Python library name strings are case insensitive which means that
-    libraries are equivalent even if the casing of the library names are
-    different. This function normalizes python library names such that
-    identical libraries have the same library name strings.
+    """Returns a normalized version of the python library name.
+
+    Normalization of a library name means converting the library name to
+    lowercase, and removing any "[...]" suffixes that occur. The reason we do
+    this is because of 2 potential confusions when comparing library names that
+    will cause this script to find incorrect mismatches.
+
+    1. Python library name strings are case insensitive which means that
+       libraries are equivalent even if the casing of the library names are
+       different.
+    2. There are certain python libraries with a default version and multiple
+       variants. These variants have names like `library[sub-library]` and
+       signify that it is a version of the library with special support for
+       the sub-library.
```

Comment:
Needs more explanation on why items with different [...] suffixes should still be considered equivalent.

Reply:
Done.